apps/aehttp/test/aehttp_ws_test_utils.erl

```diff
@@ -0,0 +1,155 @@
+-module(aehttp_ws_test_utils).
+-behaviour(websocket_client).
+
+%% API
+-export([start/2,
+         send/4,
+         subscribe_for_event/3,
+         unsubscribe_for_event/3,
+         wait_for_event/2,
+         wait_for_event/3,
+         stop/1]).
+
+%% behaviour exports
+-export([
+         init/1,
+         onconnect/2,
+         ondisconnect/2,
+         websocket_handle/3,
+         websocket_info/3,
+         websocket_terminate/3
+        ]).
+
+start(Host, Port) ->
+    crypto:start(),
+    ssl:start(),
+    WsAddress = "ws://" ++ Host ++ ":" ++ integer_to_list(Port) ++ "/websocket",
+    ct:log("connecting to ~p", [WsAddress]),
+    {ok, Pid} = websocket_client:start_link(WsAddress, ?MODULE, self()),
+    ok = wait_for_event(websocket, connected),
+    {ok, Pid}.
+    
+
+stop(ConnPid) ->
+    ct:log("Websocket disconnecting"),
+    ConnPid ! stop.
+
+send(ConnPid, Target, Action, Payload) ->
+    Msg0 = #{target => Target,
+             action => Action},
+    Msg =
+        case Payload == [] orelse Payload == #{} of
+            true -> Msg0;
+            false -> maps:put(payload, Payload, Msg0)
+        end,
+    ct:log("Sending to server ~p", [Msg]),
+    websocket_client:cast(ConnPid, {text, jsx:encode(Msg)}),
+    ok.
+
+subscribe_for_event(ConnPid, Origin, Action) ->
```

Comment:
Could you please add a line of comment mentioning that caller will receive in mailbox messages looking like {websocket_event, ...} (or similar - better if with a type).